### PR TITLE
TypeScript: Layout Provider from `story-editor` package

### LINKED ISSUE #12742: TypeScript: Convert layout provider

### PR DESCRIPTION
**Pull request by miina**
## To-do

- [x] Dependence of `themeHelpers`. 

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #12742

### CODE REVIEW

File: packages/story-editor/src/app/layout/useZoomSetting.ts
Comment (miina) [inline packages/story-editor/src/app/layout/useZoomSetting.ts:29]: The `themeHelpers` from the `design-system` above is used only for `themeHelpers.SCROLLBAR_WIDTH` constant, nothing else.
It's a bit much to type the whole `themeHelpers` within this PR since it would probably require typing `theme`, too. 

We could mark this PR blocked by #12659 or we could temporarily export the `SCROLLBAR_WIDTH` as a separate constant from somewhere and use that directly instead of `themeHelpers`. A constant isn't a helper anyway really, I guess. 

Thoughts? @barklund @swissspidy

File: packages/story-editor/src/components/footer/carousel/carouselLayout.js
Comment (barklund) [inline packages/story-editor/src/components/footer/carousel/carouselLayout.js]:
```suggestion
    CarouselState.Opening,
```

File: packages/story-editor/src/types/layoutProvider.ts
Comment (barklund) [inline packages/story-editor/src/types/layoutProvider.ts]: But we always have state first, actions second! 🥺

**Review by barklund — APPROVED** [review]
Looks great, just a nit (and some ESLint issues).

**Comment by miina** [comment]
The Karma failure is valid, looking into it...

**Comment by miina** [comment]
Karma failures are due to this only after re-running, merging:
`[percy] Error: Can only finalize pending builds`

**Pull request merged**
